core/store/src/config.rs

```diff
@@ -176,3 +189,11 @@ impl Default for MigrationSnapshot {
         Self::Enabled(true)
     }
 }
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct TrieCacheConfig {
+    /// Limit the number trie nodes that fit in the cache.
+    pub max_entries: Option<u64>,
+    /// Limit the sum of all value sizes that fit in the cache.
+    pub max_bytes: Option<u64>,
```

Comment:
TBH, after the investigation into trie sieze which showed that we have about 100 bytes-per-entry overhead, I am leaning stronger that we should configure *just* the size in bytes, and limit the number of entries by saying that each entry is 100 bytes + val.len() in size. 

This also makes sense from the user's pov: it's clear what 2gig of cache means. IN contrast, 2mil entries is opaque.

Reply:
@Longarithm What do you think? Would it make sense to only have a configuration for total bytes and drop the number of entries config?